staging/src/k8s.io/component-base/zpages/statusz/registry.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package statusz
+
+import (
+	"time"
+
+	"github.com/blang/semver/v4"
+	"k8s.io/component-base/version"
+	"k8s.io/klog/v2"
+
+	"k8s.io/component-base/featuregate"
+	compbasemetrics "k8s.io/component-base/metrics"
+)
+
+type statuszRegistry interface {
+	processStartTime() time.Time
+	goVersion() string
+	binaryVersion() semver.Version
+	emulationVersion() semver.Version
+	minCompatibilityVersion() semver.Version
+	usefulLinks() map[string]string
+}
+
+type registry struct{}
+
+func (registry) processStartTime() time.Time {
+	start, err := compbasemetrics.GetProcessStart()
```

Comment:
Instead of exposing the internal function, I'd like to propose keeping a dedicated `/statusz` registry and [extending `kubeRegistry`](https://github.com/kubernetes/kubernetes/blob/63f98e035df0e60530fb0f201ddea546ace949c5/staging/src/k8s.io/component-base/metrics/wrappers.go#L139) by incorporating [`CollectAndFormat`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/testutil#CollectAndFormat) to enable this to scale better (and be able to reuse the existing `RegisterProcessStartTime` for all fields here, and even expose them as metrics, although we may want to keep this hidden for internal use only, at first).

Reply:
Not sure I understand, why would we want to use the PromRegistry here? This is not related to metrics at all?

Reply:
Backing `/statusz` data by a registry will give the users a way to consume this data through their metrics-backend while allowing us to reuse the existing machinery to output that information to the page.

Reply:
> Backing /statusz data by a registry will give the users a way to consume this data through their metrics-backend

Oh, yeah the consensus from sig-arch was not to make this endpoint be machine parseable in alpha (at least not until we have done the API-fication for these z-endpoints that would make these safe to be scraped)

But thanks for the ideas about integrating these with metrics backend as a possibility in future, I will keep this comment unresolved for future iterations regarding making this endpoint machine readable.

Reply:
> not to make this endpoint be machine parseable

So, assuming my suggestion above goes through, we'd most likely be emitting those metrics from `/metrics` rather than `/statusz`.